File: sorting/archived/selection.py
def selection_sort(arr):
    for i in range(len(arr) - 1):
        idx = i
        for j in range(idx + 1, len(arr)):
            if arr[j] < arr[idx]:
                idx = j
        arr[i], arr[idx] = arr[idx], arr[i]
    return arr

print(selection_sort([3, -2, 9, 0, 12, -5, 8, 0])) # [-5, -2, 0, 0, 3, 8, 9, 12]

File: src/sorting/selection.rs
pub fn selection_sort<T: Ord + Copy>(arr: &mut [T]) {
    let len: usize = arr.len();
    for i in 0..len {
        let idx: usize = (i..len).min_by_key(|x| arr[*x]).unwrap();
        arr.swap(i, idx);
    }
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn selection_sort_test() {
        let mut arr: Vec<i32> = vec![3, -2, 9, 0, 12, -5, 8, 0];
        selection_sort(&mut arr);
        assert_eq!(arr, vec![-5, -2, 0, 0, 3, 8, 9, 12]);
    }
}
